tgui/packages/tgui/interfaces/GeneTek.js

```diff
@@ -0,0 +1,1316 @@
+  const { data, act } = useBackend(context);
+  const {
+    activeGene,
+  } = data;
+  const {
+    genes,
+    noSelection,
+    noGenes,
+    ...rest
+  } = props;
+  const ag = genes.find(g => g.ref === activeGene);
+
+  return (
+    <Fragment>
+      <Flex wrap mb={1}>
+        {genes.map(g => (
+          <Flex.Item
+            key={g.ref}
+            grow={1}
+            textAlign="center">
+            <Button
+              icon={g.research >= 2 ? "flask"
+                : g.research >= 1 ? "hourglass"
+                  : "question"}
+              color={g.ref === activeGene ? "black"
+                : g.research >= 3 ? "good"
+                  : g.research >= 2 ? "teal"
+                    : g.research >= 1 ? "average"
+                      : "grey"}
+              onClick={() => act("setgene", { ref: g.ref })}
+              tooltip={g.research === 1 ? "Researching..." : g.name}
+              tooltipPosition="left"
+              width="80%" />
+          </Flex.Item>
+        ))}
+      </Flex>
+      {genes.length ? ag ? (
+        <BioEffect
+          key={ag.ref}
+          gene={ag}
+          {...rest} />
+      ) : (noSelection || "Select a gene to view it.")
+        : (noGenes || "No genes found.")}
+    </Fragment>
+  );
+};
+
+const Description = (props, context) => {
+  const lines = props.text.split(/<br ?\/?>/g);
+
+  return lines.map((line, i) => (
+    <p key={i}>
+      {line}
+    </p>
+  ));
+};
+
+const BioEffect = (props, context) => {
+  const { data, act } = useBackend(context);
+  const [booth, setBooth] = useSharedState(context, "booth", null);
+  const {
+    materialCur,
+    researchCost,
+    equipmentCooldown,
+    saveSlots,
+    savedMutations,
+    haveSubject,
+    subjectStat,
+    boothCost,
+    injectorCost,
+    precisionEmitter,
+    toSplice,
+  } = data;
+  const {
+    gene,
+    isSample,
+    isPotential,
+    isActive,
+    isStorage,
+  } = props;
+  const {
+    ref,
+    name,
+    desc,
+    icon,
+    research,
+    canResearch,
+    canInject,
+    canScramble,
+    canReclaim,
+    spliceError,
+    dna,
+  } = gene;
+
+  const haveDevice = {
+    Injectors: false,
+    Analyzer: false,
+    Emitter: false,
+    Reclaimer: false,
+  };
+  const onCooldown = {
+    Injectors: true,
+    Analyzer: true,
+    Emitter: true,
+    Reclaimer: true,
+  };
+  for (const { label, cooldown } of equipmentCooldown) {
+    haveDevice[label] = true;
+    onCooldown[label] = cooldown > 0;
+  }
+
+  const dnaGood = dna.every(pair => !pair[2]);
+  const dnaGoodExceptLocks = dna.every(pair =>
+    !pair[2] || pair[3] === "locked");
+
+  return (
+    <Section
+      title={name}
+      buttons={
+        <GeneIcon
+          name={icon}
+          size={1.5} />
+      }>
+      {!!booth && booth.ref === ref && (
+        <Modal>
+          <Section
+            width={35}
+            title={name}
+            style={{
+              "margin": "-10px",
```

Comment:
`-10px` margin seems pretty interesting. What's this for?

Reply:
It makes the modal not have a huge amount of padding around the section. Should probably move that into a stylesheet so everyone gets the benefit.